src/combinators/mod.rs

```diff
@@ -406,3 +406,115 @@ where
         }
     })
 }
+
+/// Return an eventual with a starting value that then defers to source.
+pub fn init_with<R>(source: R, value: R::Output) -> Eventual<R::Output>
+where
+    R: IntoReader,
+{
+    let mut source = source.into_reader();
+    Eventual::spawn(|mut writer| async move {
+        writer.write(value);
+        loop {
+            writer.write(source.next().await?);
+        }
+    })
+}
+
+/// Prefer values from source_1 if available, otherwise use source_2.
+pub fn prefer<R1, R2, T>(source_1: R1, source_2: R2) -> Eventual<T>
+where
+    R1: IntoReader<Output = T>,
+    R2: IntoReader<Output = T>,
+    T: Value,
+{
+    let mut source_1 = source_1.into_reader();
+    let mut source_2 = source_2.into_reader();
+
+    Eventual::spawn(|mut writer| async move {
+        loop {
+            select! {
+                biased;
+
+                one = source_1.next() => {
+                    if let Ok(one) = one {
+                        writer.write(one);
+                        break;
+                    } else {
+                        loop {
+                            writer.write(source_2.next().await?);
```

Comment:
Shouldn't we `drop(source_1)` here?

Reply:
Correct me if I'm wrong, but I don't think that accomplishes anything.

The reason for `drop(source_2)` on line 458 is to allow writes to cancel. On line 445 `source_1` has already returned `Err(Closed)` so this concern does not apply.

Reply:
Ok, I see. The drop below is for the break on line 442, not the one on line 453. I understand now.